src/helm/proxy/clients/huggingface_client.py

```diff
@@ -29,27 +30,30 @@ def __init__(self, model_config: HuggingFaceModelConfig):
         model_kwargs = {}
         if model_config.revision:
             model_kwargs["revision"] = model_config.revision
+        # If the HuggingFace model is stored locally, it will have a path defined and we should load it from there.
+        # Otherwise, download it from the HuggingFace hub by passing in its identifier.
+        if model_config.path:
+            model_name = model_config.path
+        else:
+            model_name = model_config.model_id
         with htrack_block(f"Loading Hugging Face model for config {model_config}"):
             self.model = AutoModelForCausalLM.from_pretrained(
-                model_config.model_id, trust_remote_code=True, **model_kwargs
+                model_name, trust_remote_code=True, **model_kwargs
             ).to(self.device)
         with htrack_block(f"Loading Hugging Face tokenizer model for config {model_config}"):
-            self.tokenizer = AutoTokenizer.from_pretrained(model_config.model_id, **model_kwargs)
+            self.tokenizer = AutoTokenizer.from_pretrained(model_name, **model_kwargs)
 
     def serve_request(self, raw_request: Dict[str, Any]):
-        encoded_input = self.tokenizer(raw_request["prompt"], return_tensors="pt").to(self.device)
+        encoded_input = self.tokenizer(raw_request["prompt"], return_tensors="pt", return_token_type_ids=False).to(self.device)
         raw_request = deepcopy(raw_request)
         raw_request["do_sample"] = True
         raw_request["return_dict_in_generate"] = True
         raw_request["output_scores"] = True
         top_k_per_token: int = raw_request["top_k_per_token"]
         del raw_request["top_k_per_token"]
         if len(raw_request["stop_sequences"]) > 0:
-            stop_sequence_ids = self.tokenizer(raw_request["stop_sequences"])
-            # Total number of stop words should be 1.
-            assert len(stop_sequence_ids.input_ids) == 1
-            # Total number of tokens in each stop word should be 1.
-            assert len(stop_sequence_ids.input_ids[0]) == 1
```

Comment:
Why is the last `assert` removed?

Reply:
Some tokenizers, like `LlamaTokenizer`, always add BOS and EOS tokens when they tokenize. So, when we tokenize the stop sequence here, it ends up having 3 tokens. So far no errors have come up related to removing this assert, so I hope removing it is okay